physical/shuffle.go

```diff
@@ -0,0 +1,80 @@
+package physical
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/pkg/errors"
+
+	"github.com/cube2222/octosql/execution"
+	"github.com/cube2222/octosql/graph"
+	"github.com/cube2222/octosql/physical/metadata"
+)
+
+type ShuffleInput struct {
+	Source Node
+}
+
+type node struct {
+	ShuffleInputs []*ShuffleInput
+}
+
+type ShuffleStrategy interface {
+	CalculatePartition(record *execution.Record, outputs int) int
```

Comment:
What is the `int` result? It's not entirely clear to me since this function isn't called anywhere in the code (yet)

Reply:
The partition number, I just filled it in as a placeholder for now, this is subject to change.